viam-orb-slam3.go

```diff
@@ -489,11 +489,25 @@ func (orbSvc *orbslamService) GetSLAMProcessConfig() pexec.ProcessConfig {
 	args = append(args, "-port="+orbSvc.port)
 	args = append(args, "--aix-auto-update")
 
+	target := orbSvc.executableName
+
+	appDir := os.Getenv("APPDIR")
```

Comment:
The need to add this was found by @zaporter-work.

Without this line viam-orb-slam3 would use the carto_grpc_server binary within the unix user's path, not the one shipped within the app image (which is the one we want). This env var points to the filesystem subtree of the app image https://docs.appimage.org/packaging-guide/environment-variables.html & is equal to something like `/tmp/.mount_orb-slfcenhr/usr/bin/orb_grpc_server`.

The reason RPlidar & RDK don't have it is b/c they don't ship any other executables within their app images.

Reply:
@Otterverse is what we are doing here the recommended approach for app images that need to spawn go executables?

Reply:
What you're doing works correctly for our AppImage build process, and as you're only going to be executing this one, specific binary, I think this is fine. The "bash" method could be more useful if you're going to be running arbitrary executable or something, but isn't needed here I'd think.

Reply:
Sounds good, thanks for explaining @Otterverse! 

@edaniels does that address your concern/question?

Reply:
Thanks both for explaining! Summarizing my understanding for both this and the thread below:

The appimage builder uses LD_PRELOAD to patch the binaries being included in the appimage.

One of these patches is to libc, to essentially add the mounted appimage directory to the user's PATH, so that any executables within the appimage can be found. However, since Go implements syscalls directly rather than going through libc, this patch has no effect in a Go binary. Therefore, we specify the absolute path to carto_grpc_server here.

Another patch is to remove the first slash from /lib/ld-linux-x86_64.so.2 (the dynamic linker) so that it becomes a relative path. As part of starting up carto_grpc_server, the dynamic linker is used to load all dynamic libraries needed by carto_grpc_server. We need to change directory to the appimage directory in order to use the dynamic linker shipped within the appimage.